File: Behavioral/Observer/Java/Subscriber.java
// Subscriber.java

public class Subscriber implements Observer {
  private String name;

  public Subscriber(String name) {
    this.name = name;
  }

  protected String getName() {
    return name;
  }

  @Override
  public void update(String news) {
    System.out.println(name + " received news update: " + news);
  }
}

File: Behavioral/Observer/JavaScript/subscriber.js
// subscriber.js

const Observer = require("./observer");

class Subscriber extends Observer {
  constructor(name) {
    super();
    this.name = name;
  }

  update(news) {
    console.log(`${this.name} received news update: ${news}`);
  }
}

module.exports = Subscriber;
